frontend/taipy-gui/src/utils/formatConversion.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2021-2024 Avaiga Private Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+/**
+ * Regular expressions used for parsing sprintf format strings.
+ */
+const re = {
+    text: /^[^\x25]+/,                         // Matches non-placeholder text
+    modulo: /^\x25{2}/,                        // Matches the '%%' escape sequence
+    placeholder: /^\x25?(?:\.(\d+))?([b-giostuvxX])/, // Matches placeholders
+};
+
+/**
+ * Parses a sprintf format string and returns a parse tree.
+ * @param fmt The sprintf format string.
+ * @returns The parse tree representing the structure of the format string.
+ */
+const sprintf_parse = (fmt: string) => {
+    let _fmt = fmt;
+    let match;
+    const parse_tree = [];
+
+    while (_fmt) {
+        if ((match = re.text.exec(_fmt)) !== null) {
+            // Non-placeholder text
+            parse_tree.push(match[0]);
+        } else if ((match = re.modulo.exec(_fmt)) !== null) {
+            // '%%' escape sequence
+            parse_tree.push('%');
+        } else if ((match = re.placeholder.exec(_fmt)) !== null) {
+            // Placeholder
+            if (match && match[0]) {
+                parse_tree.push({
+                    placeholder: match[0],
+                });
+            }
+        }
+
+        if (match) {
+            _fmt = _fmt.substring(match[0].length);
+        }
+    }
+
+    return parse_tree;
+}
+
+/**
+ * Converts sprintf format string to a D3 format string.
+ * @param format The sprintf format string.
+ * @returns The D3 format string.
+ */
+export const sprintfToD3Converter = (format: string) => {
+    const precisionFormat = (precision: string | undefined, specifier: string) => {
```

Comment:
this could be externalize from `sprintfToD3Converter`

Reply:
resolved